src/lib/use-is-keyboard-shown.ts

```diff
@@ -0,0 +1,29 @@
+import React from 'react';
+import { Keyboard } from 'react-native';
+
+export const useIsKeyboardShown = () => {
+  const [isKeyboardVisible, setKeyboardVisible] = React.useState(false);
+
+  React.useEffect(() => {
+    const keyboardDidShowListener = Keyboard.addListener(
+      'keyboardDidShow',
+      () => {
+        setKeyboardVisible(true); // or some other action
+      },
+    );
+
+    const keyboardDidHideListener = Keyboard.addListener(
+      'keyboardDidHide',
+      () => {
+        setKeyboardVisible(false); // or some other action
+      },
+    );
+
+    return () => {
+      keyboardDidHideListener.remove();
+      keyboardDidShowListener.remove();
```

Comment:
Maaan these look similar these lines

Reply:
Maybe ShowListener & HideListener for naming. My dyslexic mind cannot parse this.

HideListener.remove()
ShowListener.remove()


More readable?

Reply:
Good point, went with your suggestion!